packages/compartment-mapper/test/test-error-handling.js

```diff
@@ -30,6 +30,12 @@ const onError = (t, { error, title }) => {
   } else {
     t.fail();
   }
+
+  // Skip snapshotting the 'both' case because esm/csj order is not deterministic 
+  // which causes the error to contain one or two causes.
+  if(title.match(/both/i)) {
+    return t.pass();
+  }
```

Comment:
The inconsistent behavior happens when importing the same specifier from both cjs and esm modules when there's no package matching the specifier.
The order in which we attempt importing dependencies is not guaranteed.

    If we require the missing package from cjs first, the error gets postponed to runtime, but it doesn't get cached so importing from an esm module errors out before the application runs and that's the single error.
    If we import esm first, the error happens right away and gets cached so the subsequent cjs require of the missing package gets the error right away instead of postponing to runtime.

Reply:
It is fluctuating when running the same exact application via the same exact means. In both cases the error will prevent the application from working because an error is thrown and the only difference is in aggregation of errors - we either get one or two.  
An app with that issue does not reach runtime. 
The non-error case of this was a known issue when I approached the CJS support improvements - that Endo differs from other engines in unifying the module loaded from CJS and ESM into one. 
That unification has a gap - CJS require deferred error is wrapped in a module record and it makes it not-the-same as the errored record for the missing module that we got from importing.  
When we discussed deferredError, applying it to ESM was deemed a waste of good fail-fast and I agree with that.

I'll give it a second thought, but other than confusing the user with 1 or 2 "underlying errors" interchangeably I don't think this behavior hurts. Any result of it should not be accessible at runtime, because the ESM import errors out. The only thing I have to think about is if catching the error from require in CJS and reacting to it is possible.  
At worst, if all is spawned from CJS and both paths to the missing module are wrapped in separate try-catch it might be a source of 1 bit of randomness.

This PR did not introduce it though, I just added snapshotting to a test that was already reproducing the situation, but we didn't check in enough detail.
deferredError being applied to only the CJS and not ESM is the root cause.

Reply:
Quick note from today's meeting (recorded) 

- The same non-deterministic behavior could be observed in all-ESM case by inspecting the message from a rejected dynamic import promise for one of two nested missing packages to see which one failed first. Assuming the time-inconsistency comes from file load times or the host itself being non-deterministic in promise resolution at runtime.
- The current behavior is success-confluent
- I still intend to try to pin down the exact reason why the non-determinism is introduced
- Decreasing the likelihood of the condition occurring to less than now, but not zero is not worth pursuing.

Reply:
I’m satisfied, so resolving.